packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,311 @@
+
+// Keep this as early on the file as possible
+capture.init();
+
+// Automatically set by AWS
+const lambdaName = getEnvOrFail("AWS_LAMBDA_FUNCTION_NAME");
+const region = getEnvOrFail("AWS_REGION");
+// Set by us
+const axiosTimeoutSeconds = Number(getEnvOrFail("AXIOS_TIMEOUT_SECONDS"));
+const bucketName = getEnvOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+const envType = getEnvOrFail("ENV_TYPE");
+// converter config
+const FHIRToCDAConverterUrl = getEnvOrFail("FHIR_TO_CDA_CONVERTER_URL");
+const convertDocLambda = getEnvOrFail("CONVERT_DOC_LAMBDA_NAME");
+const converterKeysTableName = getEnvOrFail("SIDECHAIN_FHIR_CONVERTER_KEYS_TABLE_NAME");
+
+const lambdaClient = makeLambdaClient(region);
+const s3Client = makeS3Client(region);
+const isSandbox = envType === "sandbox";
+
+export const handler = Sentry.AWSLambda.wrapHandler(async (req: FhirToMedicalRecordPayload) => {
+  const { bundle, patientId, firstName, cxId, resources, dateFrom, dateTo, conversionType } = req;
+
+  console.log(
+    `Running with conversionType: ${conversionType}, patientId: ${patientId}, cxId: ${cxId}, resources: ${resources}, dateFrom: ${dateFrom}, dateTo: ${dateTo}`
+  );
+
+  console.log(`bundle: ${JSON.stringify(bundle)}`);
+
+  try {
+    const log = prefixedLog(`patient ${patientId}`);
+
+    if (isSandbox) {
+      const convertUrl = await convertDoc({ fileName: `${firstName}-consolidated.xml`, conversionType });
+
+      return convertUrl;
+    }
+
+    const res = await postToConverter({
+      url: FHIRToCDAConverterUrl,
+      payload: bundle,
+      converterKeysTableName,
+      axiosTimeoutSeconds,
+      log,
+    });
+
+    const formattedXML = formatXML(res.data);
+    const fileName = createFileName({ cxId, patientId, resources, dateFrom, dateTo });
+
+    await s3Client
+      .putObject({
+        Bucket: bucketName,
+        Key: fileName,
+        Body: formattedXML,
+        ContentType: "application/xml",
+      })
+      .promise();
+
+    const convertUrl = await convertDoc({ fileName, conversionType });
+
+    return convertUrl;
+  } catch (err) {
+    console.log(
+      `Error processing bundle for patient: ${patientId} with resources ${resources}; ${err}`
+    );
+    capture.error(err, {
+      extra: {
+        error: err,
+        patientId,
+        resources,
+        dateFrom,
+        dateTo,
+        context: lambdaName,
+      },
+    });
+    throw err;
+  }
+});
+
+const formatXML = (xml: string): string => {
+  const parser = new DOMParser();
+  const document = parser.parseFromString(xml, "text/xml");
+
+  const structuredBody = document.getElementsByTagName("structuredBody")[0];
+
+  const components = structuredBody
+    ? Array.from(structuredBody.getElementsByTagName("component"))
+    : [];
+
+  const problemListComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0]?.nodeValue === "Problem list";
```

Comment:
for a sanity check - will this always be capitalized as such? ie never "Problem List"?

if we're not sure, should we `.toLower()` before doing the comparison?